src/communication_managers/iothub_communication_manager/src/iothub_communication_manager.c

```diff
@@ -0,0 +1,714 @@
+ * If the connection string contains the DeviceId it is an ADUC_ConnType_Device
+ * If the connection string contains the DeviceId AND the ModuleId it is an ADUC_ConnType_Module
+ * @param connectionString the connection string to scan
+ * @returns the connection type for @p connectionString
+ */
+static ADUC_ConnType GetConnTypeFromConnectionString(const char* connectionString)
+{
+    ADUC_ConnType result = ADUC_ConnType_NotSet;
+
+    if (connectionString == NULL)
+    {
+        Log_Debug("Connection string passed to GetConnTypeFromConnectionString is NULL");
+        return ADUC_ConnType_NotSet;
+    }
+
+    if (ConnectionStringUtils_DoesKeyExist(connectionString, "DeviceId"))
+    {
+        if (ConnectionStringUtils_DoesKeyExist(connectionString, "ModuleId"))
+        {
+            result = ADUC_ConnType_Module;
+        }
+        else
+        {
+            result = ADUC_ConnType_Device;
+        }
+    }
+    else
+    {
+        Log_Debug("DeviceId not present in connection string.");
+    }
+
+    return result;
+}
+
+/**
+ * @brief Get the Connection Info from connection string, if a connection string is provided in configuration file
+ *
+ * @return true if connection info can be obtained
+ */
+static bool GetConnectionInfoFromConnectionString(ADUC_ConnectionInfo* info, const char* connectionString)
+{
+    bool succeeded = false;
+    if (info == NULL)
+    {
+        goto done;
+    }
+    if (connectionString == NULL)
+    {
+        goto done;
+    }
+
+    memset(info, 0, sizeof(*info));
+
+    char certificateString[8192];
+
+    if (mallocAndStrcpy_s(&info->connectionString, connectionString) != 0)
+    {
+        goto done;
+    }
+
+    info->connType = GetConnTypeFromConnectionString(info->connectionString);
+
+    if (info->connType == ADUC_ConnType_NotSet)
+    {
+        Log_Error("Connection string is invalid");
+        goto done;
+    }
+
+    info->authType = ADUC_AuthType_SASToken;
+
+    // Optional: The certificate string is needed for Edge Gateway connection.
+    ADUC_ConfigInfo config = {};
+    if (ADUC_ConfigInfo_Init(&config, ADUC_CONF_FILE_PATH) && config.edgegatewayCertPath != NULL)
+    {
+        if (!LoadBufferWithFileContents(config.edgegatewayCertPath, certificateString, ARRAY_SIZE(certificateString)))
+        {
+            Log_Error("Failed to read the certificate from path: %s", config.edgegatewayCertPath);
+            goto done;
+        }
+
+        if (mallocAndStrcpy_s(&info->certificateString, certificateString) != 0)
+        {
+            Log_Error("Failed to copy certificate string.");
+            goto done;
+        }
+
+        info->authType = ADUC_AuthType_NestedEdgeCert;
+    }
+
+    succeeded = true;
+
+done:
+    ADUC_ConfigInfo_UnInit(&config);
+    return succeeded;
+}
+
+/**
+ * @brief Get the Connection Info from Identity Service
+ *
+ * @return true if connection info can be obtained
+ */
+static bool GetConnectionInfoFromIdentityService(ADUC_ConnectionInfo* info)
```

Comment:
nit: For all these functions but IotHub_CommunicationManager_<name> so it's easier to reference